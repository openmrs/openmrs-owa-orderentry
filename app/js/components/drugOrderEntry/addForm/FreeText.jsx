import React from 'react';
import PropTypes from 'prop-types';

const FreeText = ({
  fieldErrors,
  fields,
  options,
  handleChange,
  handleSubmit,
  handleCancel,
  handleValidation,
  activateSaveButton,
}) => {
  const { drugInstructions, dispensingQuantity, dispensingUnit } = fields;
  return (
    <div>
      <form className="simple-form-ui">
        <div className="flex-row">
          <div>
            <label name="notes">Complete Instructions</label>
            <textarea
              rows="4"
              cols="80"
              id="notes"
              name="drugInstructions"
              value={drugInstructions}
              onChange={handleChange}
            />
          </div>
        </div>
        <div className="flex-row">
          <div>
            <p className="left label-margin">
              <label>Dispense:</label>
            </p>
            <p className="left p-margin">
              <input
                className={`medium-input ${fieldErrors.dispensingQuantity && "illegalValue"}`}
                placeholder="Quantity"
                name="dispensingQuantity"
                id="dispenseQuantity"
                type="number"
                min="0"
                onBlur={handleValidation}
                onChange={handleChange}
                value={fields.dispensingQuantity} />
              {
                fieldErrors.dispensingQuantity ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left">
              <input
                className={(fields.dispensingQuantity && !fields.dispensingUnit) ? "illegalValue" : ""}
                placeholder="Units"
                name="dispensingUnit"
                id="drugDispensingUnits"
                list="dispensingUnits"
                size="20"
                value={fields.dispensingUnit}
                onBlur={handleValidation}
                onChange={handleChange} />
              <datalist id="dispensingUnits" >
                {
                  options.dispensingUnit &&
                    options.dispensingUnit.map(unit =>
                      (
                        <option key={unit.uuid} value={unit.display} />
                      ))
                }
              </datalist>
              {
                (fields.dispensingQuantity && !fields.dispensingUnit) &&
                  <span className="field-error">Required</span>
              }
            </p>
          </div>
        </div>
        <div className="flex-row">
          <div>
            <button
              type="button"
              className="cancel"
              onClick={handleCancel}
            >
            Cancel
            </button>
          </div>
          <div className="column-1">
            <button
              type="button"
              className="confirm"
              onClick={handleSubmit}
              disabled={activateSaveButton()}
            >
            Add
            </button>
          </div>
        </div>
      </form>
    </div>
  );
};

FreeText.propTypes = {
  fieldErrors: PropTypes.object.isRequired,
  handleValidation: PropTypes.func.isRequired,
  activateSaveButton: PropTypes.func.isRequired,
  fields: PropTypes.object.isRequired,
  options: PropTypes.object,
  handleCancel: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

FreeText.defaultProps = {
  options: {},
};

export default FreeText;
