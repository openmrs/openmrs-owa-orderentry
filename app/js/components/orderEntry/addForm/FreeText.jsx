import React from 'react';


class FreeText extends React.Component {

  render() {
    return (
      <div className="tab-pane fade"
        id="free"
        role="tabpanel"
        aria-labelledby="free-tab">
        <form>
          <div className="row">
            <div className="col-sm-11">
              <div className="form-group">
                <label htmlFor="instructions">Completete Instructions:</label>
                <textarea className="form-control" rows="5" id="instructions" />
              </div>
            </div>
          </div>
        </form>

        <div className="row">
          <div className="col-sm-3 text-left">
            <button type="button" className="btn btn-danger">Cancel</button>
          </div>
          <div className="col-sm-3 col-sm-offset-5 text-right">
            <button type="button" className="btn btn-success">Save</button>
          </div>
        </div>
      </div>
    );
  }
}

export default FreeText;
