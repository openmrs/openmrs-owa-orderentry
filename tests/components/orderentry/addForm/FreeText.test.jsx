import React from 'react';

import FreeText from '../../../../app/js/components/orderEntry/addForm/FreeText';
import store from '../../../../app/js/redux-store';

describe('Test for Free text form', () => {
  it('should render component', () => {
    const wrapper = shallow(<FreeText />  );
    expect(wrapper).toMatchSnapshot()
  });
});
