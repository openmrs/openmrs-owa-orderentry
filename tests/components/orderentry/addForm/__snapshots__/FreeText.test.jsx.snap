// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Free text form should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FreeText />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-labelledby": "free-tab",
      "children": Array [
        <form>
          <div
            className="row"
          >
            <div
              className="col-sm-11"
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="instructions"
                >
                  Completete Instructions:
                </label>
                <textarea
                  className="form-control"
                  id="instructions"
                  rows="5"
                />
              </div>
            </div>
          </div>
        </form>,
        <div
          className="row"
        >
          <div
            className="col-sm-3 text-left"
          >
            <button
              className="btn btn-danger"
              type="button"
            >
              Cancel
            </button>
          </div>
          <div
            className="col-sm-3 col-sm-offset-5 text-right"
          >
            <button
              className="btn btn-success"
              type="button"
            >
              Save
            </button>
          </div>
        </div>,
      ],
      "className": "tab-pane fade",
      "id": "free",
      "role": "tabpanel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-sm-11"
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="instructions"
                >
                  Completete Instructions:
                </label>
                <textarea
                  className="form-control"
                  id="instructions"
                  rows="5"
                />
              </div>
            </div>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-sm-11"
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="instructions"
                >
                  Completete Instructions:
                </label>
                <textarea
                  className="form-control"
                  id="instructions"
                  rows="5"
                />
              </div>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="form-group"
              >
                <label
                  htmlFor="instructions"
                >
                  Completete Instructions:
                </label>
                <textarea
                  className="form-control"
                  id="instructions"
                  rows="5"
                />
              </div>,
              "className": "col-sm-11",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="instructions"
                  >
                    Completete Instructions:
                  </label>,
                  <textarea
                    className="form-control"
                    id="instructions"
                    rows="5"
                  />,
                ],
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Completete Instructions:",
                    "htmlFor": "instructions",
                  },
                  "ref": null,
                  "rendered": "Completete Instructions:",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "id": "instructions",
                    "rows": "5",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-sm-3 text-left"
            >
              <button
                className="btn btn-danger"
                type="button"
              >
                Cancel
              </button>
            </div>,
            <div
              className="col-sm-3 col-sm-offset-5 text-right"
            >
              <button
                className="btn btn-success"
                type="button"
              >
                Save
              </button>
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-danger"
                type="button"
              >
                Cancel
              </button>,
              "className": "col-sm-3 text-left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cancel",
                "className": "btn btn-danger",
                "type": "button",
              },
              "ref": null,
              "rendered": "Cancel",
              "type": "button",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-success"
                type="button"
              >
                Save
              </button>,
              "className": "col-sm-3 col-sm-offset-5 text-right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Save",
                "className": "btn btn-success",
                "type": "button",
              },
              "ref": null,
              "rendered": "Save",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-labelledby": "free-tab",
        "children": Array [
          <form>
            <div
              className="row"
            >
              <div
                className="col-sm-11"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="instructions"
                  >
                    Completete Instructions:
                  </label>
                  <textarea
                    className="form-control"
                    id="instructions"
                    rows="5"
                  />
                </div>
              </div>
            </div>
          </form>,
          <div
            className="row"
          >
            <div
              className="col-sm-3 text-left"
            >
              <button
                className="btn btn-danger"
                type="button"
              >
                Cancel
              </button>
            </div>
            <div
              className="col-sm-3 col-sm-offset-5 text-right"
            >
              <button
                className="btn btn-success"
                type="button"
              >
                Save
              </button>
            </div>
          </div>,
        ],
        "className": "tab-pane fade",
        "id": "free",
        "role": "tabpanel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <div
                className="col-sm-11"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="instructions"
                  >
                    Completete Instructions:
                  </label>
                  <textarea
                    className="form-control"
                    id="instructions"
                    rows="5"
                  />
                </div>
              </div>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-sm-11"
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="instructions"
                  >
                    Completete Instructions:
                  </label>
                  <textarea
                    className="form-control"
                    id="instructions"
                    rows="5"
                  />
                </div>
              </div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="form-group"
                >
                  <label
                    htmlFor="instructions"
                  >
                    Completete Instructions:
                  </label>
                  <textarea
                    className="form-control"
                    id="instructions"
                    rows="5"
                  />
                </div>,
                "className": "col-sm-11",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="instructions"
                    >
                      Completete Instructions:
                    </label>,
                    <textarea
                      className="form-control"
                      id="instructions"
                      rows="5"
                    />,
                  ],
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Completete Instructions:",
                      "htmlFor": "instructions",
                    },
                    "ref": null,
                    "rendered": "Completete Instructions:",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "id": "instructions",
                      "rows": "5",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "textarea",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-sm-3 text-left"
              >
                <button
                  className="btn btn-danger"
                  type="button"
                >
                  Cancel
                </button>
              </div>,
              <div
                className="col-sm-3 col-sm-offset-5 text-right"
              >
                <button
                  className="btn btn-success"
                  type="button"
                >
                  Save
                </button>
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-danger"
                  type="button"
                >
                  Cancel
                </button>,
                "className": "col-sm-3 text-left",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cancel",
                  "className": "btn btn-danger",
                  "type": "button",
                },
                "ref": null,
                "rendered": "Cancel",
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-success"
                  type="button"
                >
                  Save
                </button>,
                "className": "col-sm-3 col-sm-offset-5 text-right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Save",
                  "className": "btn btn-success",
                  "type": "button",
                },
                "ref": null,
                "rendered": "Save",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
